rpc/jsonrpc/client/http_json_client_test.go

```diff
@@ -84,3 +84,18 @@ func Test_parsedURL(t *testing.T) {
 		})
 	}
 }
+
+func TestMakeHTTPDialerURL(t *testing.T) {
+	remotes := []string{"https://foo-bar.com", "http://foo-bar.com"}
```

Comment:
do we want to add other kinds of protocols and/or test error cases?

Reply:
@tychoish what other kinds of protocols we will use in the HTTP client?

Reply:
I can add some error cases.

Reply:
Added 2 error test cases.